### PR TITLE
Some bugfixes

### PR DESCRIPTION
**Pull request by sklencar**
- changed TOS link
- fixed remember last value checkboxes
- notification about incorrect photo path

### CODE REVIEW

**Comment by saberraz** [comment]
Related to https://github.com/lutraconsulting/input/issues/1536

**Review by PeterPetrik — APPROVED** [review]

**Comment by jozef-budac** [comment]
Fixes were verified

File: app/qml/RegistrationForm.qml [hunk 1]
```diff
@@ -212,7 +212,7 @@ Rectangle {
         height: fieldHeight
         width: registerForm.width
         text: qsTr("I accept the Mergin %1Terms and Conditions%3 and %2Privacy Policy%3")
-              .arg("<a href='"+__merginApi.apiRoot + "assets/mergin_terms_of_service.pdf'>")
+              .arg("<a href='"+__merginApi.apiRoot + "assets/tos.html'>")
```
Comment (tomasMizera) [inline app/qml/RegistrationForm.qml:215]: There is already `__inputHelp.merginTermsLink`, let's use that one instead

File: app/qml/form/FeatureForm.qml [hunk 2]
```diff
@@ -197,7 +197,6 @@ Item {
     else
     {
       controller.create()
-      state = "Edit"
```
Comment (tomasMizera) [inline app/qml/form/FeatureForm.qml:200]: Why is this removed?

File: app/androidutils.cpp [hunk 3]
```diff
@@ -214,11 +214,8 @@ void AndroidUtils::handleActivityResult( int receiverRequestCode, int resultCode
...
     QAndroidJniObject errorJNI = data.callObjectMethod( "getStringExtra", "(Ljava/lang/String;)Ljava/lang/String;", RESULT_STRING.object<jstring>() );
     // Internal cancelation due to an error
-    if ( !errorJNI.isValid() )
-    {
-      QString errorMsg = errorJNI.toString();
-      showToast( errorMsg );
-    }
+    QString errorMsg = errorJNI.toString();
+    showToast( errorMsg );
     return;
```
Comment (tomasMizera) [inline app/androidutils.cpp:219]: Can you give some info why is this check removed now?

File: app/qml/RegistrationForm.qml (hunk 1 above)
Comment (sklencar) [reply app/qml/RegistrationForm.qml:215]: Good point! However, `__inputHelp.merginTermsLink` uses a hardcoded Mergin api root pointing at public Mergin server. If we want to support CE Mergin server, its probably better to use current client's mergin api root (`__merginApi.apiRoot`).
